libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFPushNotificationManager.m

```diff
@@ -109,6 +109,7 @@ - (void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData*)deviceTokenDat
     [SFSDKCoreLogger i:[self class] format:@"Registration with Apple for remote push notifications succeeded"];
     _deviceToken = [NSString stringWithHexData:deviceTokenData];
     [[SFPreferences currentUserLevelPreferences] setObject:_deviceToken forKey:kSFDeviceToken];
+    [[SFPreferences currentUserLevelPreferences] synchronize];
```

Comment:
It looks like both of these values are intended to be persisted based on the init below but I couldn't find any other calls to `synchronize` https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/a29fdb259318e45180e69547aefeeb054f6905eb/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFPushNotificationManager.m#L71-L75

Reply:
So they were never saved to disk? What scenario does it break? How come we didn't notice for so long?

Reply:
It looks like our sample apps always register for push notifications at start up so it would never rely on the restored value. For a local test I changed it so that I logged in -> registered -> killed the app -> launched again without registering and hit the issue, but we can also ask what the customer's flow was

Reply:
The fix makes sense. The problem probably got unnoticed because it is hard to hit.